test/unittest/tests/block/test_block_api_misc.cpp

```diff
@@ -0,0 +1,262 @@
+     *   //address cookie is valid.
+     *   REQUIRE(test_addr_invalid(s, &bm, 512, 1024, 12345) == WT_PANIC);
+     *
+     * }
+     */
+
+    // Cleanup for block created during block manager initialization.
+    REQUIRE(__wt_block_close(s, bm.block) == 0);
+    REQUIRE(__wt_block_manager_drop(s, file_path.c_str(), false) == 0);
+}
+
+TEST_CASE("Block manager addr string", "[block_api_misc]")
+{
+    // Build Mock session, this will automatically create a mock connection.
+    std::shared_ptr<mock_session> session = mock_session::build_test_mock_session();
+
+    WT_BM bm;
+    WT_CLEAR(bm);
+    auto path = std::filesystem::current_path();
+    std::string file_path(path.string() + "/test.wt");
+    setup_bm(session, &bm, file_path);
+    WT_SESSION_IMPL *s = session->get_wt_session_impl();
+
+    SECTION("Test addr string with non-zero values")
+    {
+        // (512, 1024, 12345) -> (offset, size, checksum)
+        test_addr_string(s, &bm, 512, 1024, 12345, "[0: 512-1536, 1024, 12345]");
+    }
+
+    SECTION("Test addr string with zero values")
+    {
+        // (0, 0, 0) -> (offset, size, checksum)
+        test_addr_string(s, &bm, 0, 0, 0, "[0: 0-0, 0, 0]");
+    }
+
+    SECTION("Test addr string with zero size")
+    {
+        // (512, 0, 12345) -> (offset, size, checksum)
+        test_addr_string(s, &bm, 512, 0, 12345, "[0: 0-0, 0, 0]");
+    }
+
+    REQUIRE(__wt_block_close(s, bm.block) == 0);
+}
+
+TEST_CASE("Block header", "[block_api_misc]")
+{
+    // Declare a block manager and set it up so that we can use its legal API methods.
+    WT_BM bm;
+    WT_CLEAR(bm);
+    __wti_bm_method_set(&bm, false);
+
+    SECTION("Test block header size is correct")
+    {
+        REQUIRE(bm.block_header(&bm) == (u_int)WT_BLOCK_HEADER_SIZE);
+    }
+}
+
+TEST_CASE("Block manager is mapped", "[block_api_misc]")
+{
+    // Declare a block manager and set it up so that we can use its legal API methods.
+    WT_BM bm;
+    WT_CLEAR(bm);
+    __wti_bm_method_set(&bm, false);
+
+    SECTION("Test block manager is mapped")
+    {
+        uint8_t i;
+        bm.map = &i;
+        REQUIRE(bm.is_mapped(&bm, nullptr) == true);
+    }
+
+    SECTION("Test block manager is not mapped")
+    {
+        bm.map = nullptr;
+        REQUIRE(bm.is_mapped(&bm, nullptr) == false);
+    }
+}
+
+TEST_CASE("Block manager size and stat", "[block_api_misc]")
+{
+    // Build Mock session, this will automatically create a mock connection.
+    std::shared_ptr<mock_session> session = mock_session::build_test_mock_session();
+
+    WT_BM bm;
+    WT_CLEAR(bm);
+    auto path = std::filesystem::current_path();
+    std::string file_path(path.string() + "/test.wt");
+    setup_bm(session, &bm, file_path);
+    WT_SESSION_IMPL *s = session->get_wt_session_impl();
+
+    SECTION("Test that the bm->stat method updates statistics correctly")
+    {
+        check_bm_stats(s, &bm);
+    }
+
+    wt_off_t prev_size;
```

Comment:
What is the goal for the following lines?
```
    wt_off_t prev_size;
    REQUIRE(bm.size(&bm, s, &prev_size) == 0);
```
I'm asking because it looks like they may be in the wrong location. With the current position, they will execute after the first SECTION when the test is run for that SECTION, and before the second SECTION when the test is run (a second time) for that SECTION. 

Should those lines be before the first SECTION or after the second SECTION, so they are run for each SECTION? Or should they be moved inside the second SECTION? 
either above the first SECTION or inside the following SECTION.

Reply:
you are right, it should be moved inside the second section.

Reply:
Thanks!